File: ng-shop-api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace ng_shop_api.Models
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime PlacedTime { get; set; }
        public string FullName { get; set; }
        public bool IsMale { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string Option { get; set; }
        public string ShippingAddress { get; set; }
        public string ShippingCity { get; set; }
        public string ShippingDistrict { get; set; }
        public string OrderStatus { get; set; }
        public bool IsReceivedAtStore { get; set; }
        public string StoreAddress { get; set; }

        public int? UserId { get; set; }
        public User User { get; set; }
        public ICollection<OrderDetail> OrderDetails { get; set; }
        public Order()
        {
            PlacedTime = DateTime.Now;
            OrderDetails = new Collection<OrderDetail>();
            OrderStatus = "Đã tiếp nhận";
        }
    }

    // public enum Status
    // {
    //     Received, Processing, AwaitingPickup, BeingDelivered, Completed, Cancelled
    // }
}

File: ng-shop-spa/src/app/_models/order.ts
import { User } from './user';
import { OrderDetail } from './order-detail';

export interface Order {
    id: number;
    placedTime: Date;
    fullName: string;
    isMale: boolean;
    phoneNumber: string;
    email: string;
    option: string;
    shippingAddress: string;
    shippingCity: string;
    shippingDistrict: string;
    orderStatus: string;
    isReceivedAtStore: boolean;
    storeAddress: string;

    userId?: number;
    orderDetails: OrderDetail[];
}
